cpp/src/fil/fil.cu

```diff
@@ -138,17 +133,40 @@ struct forest {
     params.data = data;
     params.num_rows = num_rows;
     params.max_shm = max_shm_;
-    params.num_output_classes = predict_proba ? 2 : 1;
+    params.num_classes = num_classes_;
+    /** FLOAT_SCALAR means inference produces 1 class score/component and
+        transform_k might complement to 2 for classification,
+        if class probabilities are being requested.
+        assuming predict(..., predict_proba=true) will not get called
+        for regression, hence predict_params::num_outputs == 2 */
+    params.num_outputs =
```

Comment:
What happens for multi-class classification when `predict_proba` is false, but `output_t::CLASS` was not used when creating the forest, so the output will still be `params.num_classes` probabilities per row?

Reply:
agreed that it should. will correct logic to match.